sql/sql_optimizer.cc

```diff
@@ -10141,6 +10141,12 @@ static bool internal_remove_eq_conds(THD *thd, Item *cond,
                                      Item::cond_result *cond_value,
                                      unsigned char part)
 {
+	bool is_cond_or = false;
+	if (dynamic_cast<Item_cond_or*>(cond)) 
```

Comment:
using `if (cond && dynamic_cast<Item_cond_or*>(cond))`

Reply:
sure,that's gook